controllers/awsmachine_controller.go

```diff
@@ -533,9 +538,11 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 		machineScope.Info("EC2 instance state changed", "state", instance.State, "instance-id", *machineScope.GetInstanceID())
 	}
 
+	shouldRequeue := false
 	switch instance.State {
 	case infrav1.InstanceStatePending:
 		machineScope.SetNotReady()
+		shouldRequeue = true
```

Comment:
I wonder. Looks like most the things after this are dependent on the instance being in a good state. ( correct me if I'm wrong I'm on mobile ). What if we would immediately return here and requeue instead of at the end? That would possibly be cleaner and would require less calls since the rest of this thing what comes after this wouldn't have to run.

Reply:
I see pending status is also defined in InstanceIsOperational and InstanceIsInKnownState. Many functions are executed against this condition.  So I didn't return it immediately, trying to not change existing workflow as far as possible. :)  

But you or @richardcase  are more familiar with the afterwards workflow and the possible impact, if you think it doesn't matter too much, it'll be worth a try. Just there might be some regression risks.

cc @yastij

Reply:
Ah, I see. You're probably right then. :)